nym-vpn-x/src-tauri/src/grpc/client.rs

```diff
@@ -224,6 +224,53 @@ impl GrpcClient {
         Ok(())
     }
 
+    /// Watch VPN status updates
+    #[instrument(skip_all)]
+    pub async fn watch_vpn_status(&self) -> Result<()> {
+        let mut vpnd = self.vpnd().await?;
+
+        let request = Request::new(Empty {});
+        let mut stream = vpnd
+            .listen_to_connection_status(request)
+            .await
+            .inspect_err(|e| {
+                error!("listen_to_connection_status failed: {}", e);
+            })?
+            .into_inner();
+
+        let (tx, mut rx) = mpsc::channel(32);
+        tokio::spawn(async move {
+            loop {
+                match stream.message().await {
+                    Ok(Some(update)) => {
+                        tx.send(update).await.unwrap();
+                    }
+                    Ok(None) => {
+                        warn!("watch vpn status stream closed by the server");
+                        return;
+                    }
+                    Err(e) => {
+                        warn!("watch vpn status stream get a grpc error: {}", e);
+                    }
+                }
+            }
+        });
+
+        while let Some(status) = rx.recv().await {
+            // TODO handle status updates
+            debug!(
+                "vpn status update {:?}, {:?}",
+                status.kind(),
+                status.message
+            );
+            if !status.details.is_empty() {
+                trace!("vpn status details: {:?}", status.details);
+            }
+        }
```

Comment:
```suggestion
 
        while let Ok(Some(status)) = stream.message().await {
            // TODO handle status updates
            debug!(
                "vpn status update {:?}, {:?}",
                status.kind(),
                status.message
            );
            if !status.details.is_empty() {
                trace!("vpn status details: {:?}", status.details);
            }
        }
```

Reply:
Thanks for the suggestion. The thing is using ` while let Ok(Some(…))` we can't handle the other possible variants i.e. `Ok(None)` or `Err(e)`, right?
Also, an `Err(e)` could be emitted one time by the server, but that does not mean the stream has finished and closed. But using the suggested `while let` pattern it will be terminated after the first non Ok(Some(..)) encoutered. Which is quite "fragile".
Even if we are just logging (for now) when these other variants kick in, I prefer to keep that for better reliability and better control on code flow.

Reply:
I will merge it as is, but I'm open to continue the discussion internally if you think it's needed.

Reply:
If you want to handle the other states you can e.g replace the while loop with `loop { select { .. } }`. The main point imho is that there is no need to spawn a separate task, since this function will anyway block while listening for messages